src/api/pokemonService.ts

```diff
@@ -0,0 +1,33 @@
+import { HttpClient } from "./httpClient";
+import { Pokemon } from "../types/Pokemon";
+import { LIMIT, ERROR_MESSAGES, NETWORK_DELAY } from "../helpers/constants";
+import { simulateError, delay } from "../helpers/apiUtils"; 
+
+//  This 'service' returns my main fetch function.
+export const createPokemonService = (httpClient: HttpClient) => {
+  const fetchPokemons = async (offset: number): Promise<Pokemon[]> => {
+    try {
+      await delay(NETWORK_DELAY);
+      simulateError();
+
+      const response = await httpClient.get<{
+        results: { name: string; url: string }[];
+      }>(`?offset=${offset}&limit=${LIMIT}`);
+      const results = response.data.results;
```

Comment:
Sempre uma boa ideia extrair tipos declarados inline, pra facilitar de entender e reaproveitar:
```suggestion
      type PokemonApi = {
          name: string;
          url: string
        }

      type ApiResults = {
        results: PokemonApi[]
      }

      const response = await httpClient.get<ApiResults>(`?offset=${offset}&limit=${LIMIT}`);
      const results = response.data.results;
```